src/aerialmap_display.cpp

```diff
@@ -466,56 +479,105 @@ void AerialMapDisplay::assembleScene()
   tileCache_.purge({ tileId, blocks_ });
 }
 
+namespace
+{
+// rotation matrix for rotating the current frame to ENU
+Ogre::Matrix3 rotationToENU(frame_convention convention)
+{
+  switch (convention)
+  {
+    case frame_convention::ned:
+    {
+      // XYZ->NED will cause the map to appear reversed when viewed from above (from +z).
+      // clang-format off
+      Ogre::Matrix3 const xyz_R_ned(0, 1, 0,
+                                    1, 0, 0,
+                                    0, 0, -1);
+      // clang-format on
+      return xyz_R_ned.Transpose();
+    }
+    case frame_convention::nwu:
+    {
+      // clang-format off
+      Ogre::Matrix3 const xyz_R_nwu(0, -1, 0,
+                                    1, 0, 0,
+                                    0, 0, 1);
+      // clang-format on
+      return xyz_R_nwu.Transpose();
+    }
+    case frame_convention::enu:
+    default:  // this should never be the case
+      return Ogre::Matrix3::IDENTITY;
+  }
+}
+}  // namespace
+
 void AerialMapDisplay::transformAerialMap()
 {
+  // We will use five frames in this function:
+  //
+  // * The frame from the NavSatFix message. It is rigidly attached to the robot.
+  // * Rviz's fixed frame (referred to as Ff). It is set by the user in Rviz.
+  // * The ENU/ NED/ NWU world frame "map". The code only uses ENU internally and rotates the frame into NED or NWU at
+  //   the end of the calculations.
+  // * The frame of the tiles. We assume that the tiles are in a frame where x points eastwards and y southwards. This
```

Comment:
a ref to mercator would be nice, defining the tile x and y coordinate system

Reply:
Done